### PR TITLE
Main page

### PR DESCRIPTION
**Pull request by vyomea**
Set up the components for main page:
Posts (post made by other users)
User posts (which are posts made by the user logged in)

Also, set up the basic layout for the main page

### CODE REVIEW

File: src/app/src/components/Post.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
+import React from "react";
+import Button, { ButtonProps } from "@mui/material/Button";
+import { styled as Styled } from "@mui/material/styles";
+import styled from "styled-components";
+import logo from "../logo.svg";
+
+interface postItem {
+  Name: String;
```
Comment (Sutanshu) [inline src/app/src/components/Post.tsx]: Suggest changing every "String" type to "string" type

See: [https://stackoverflow.com/questions/14727044/what-is-the-difference-between-types-string-and-string](url)

File: src/app/src/components/UserPost.tsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  return (
+    <PostContainer>
+      <PostProfilePictureContainer>
+        <img src={logo} alt="Profile" height="100" width="100" />
+      </PostProfilePictureContainer>
+      <PostDetailsContainer>
+        <TopRowContainer>
+          <NameContainer>{props?.Name}</NameContainer>
+          <EditDeleteButtonContainer>
+            <EditButton>Edit</EditButton>
```
Comment (Sutanshu) [inline src/app/src/components/UserPost.tsx:125]: Can implement Edit button functionality by using states

**Comment by Sutanshu** [comment]
There's this weird spacing between the nav bar and the posts, doesn't seem to be consistent with the screen mockups
Screen Shot 2022-02-20 at 7 29 21 PM

File: src/app/src/components/Post.tsx (hunk 1 above)
Comment (vyomea) [reply src/app/src/components/Post.tsx]: Changed

**Comment by vyomea** [comment]
> There's this weird spacing between the nav bar and the posts, doesn't seem to be consistent with the screen mockups Screen Shot 2022-02-20 at 7 29 21 PM

Yeah, the margin is fixed for now, so the spacing you see will be different for screens with different resolutions

**Review by Sutanshu — APPROVED** [review]

File: src/app/src/components/UserPost.tsx (hunk 2 above)
Comment (vyomea) [reply src/app/src/components/UserPost.tsx:125]: We can add stated later imo.

**Pull request merged**